### DISCUSSION TITLE
Official support For "Ersatzcheckins"

### DISCUSSION THREAD

**Discussion by ChillLP** [Traewelling/traewelling#2039] [Ideas / Feature Requests]
For example the RB38, RB24 and so on to Köln Messe/Deutz cant be found in Hafas from Cologne Central, so an offiziell Support For Ersatzcheckins would be nice.

What i envision is i can click on an RE1 For example, and in the checkin For that train I can click a chrckbox that activates a dropdown and a Textfeld where i can Put In the Type of train (Express, Regional, Subway, Special service or even maybe Cargo train) and in the Textfeld i can Put the trainnumber. Regarding csrgotrains, For example 51503 Runs from Köln Gremberg to Hagen Vorhalle, and between Wuppertal and Hagen i could Check-in on a S8 but using the ersatzcheckin i could correct my Data to say that im on a Cargo train. Some Special Services are also Not to be found in Hafas i think Like Fußballsonderzüge or Leerzüge.

**Comment by ChillLP** [comment]
![image](https://user-images.githubusercontent.com/105374102/222980117-b4f378e1-708a-47bd-bfc4-99c6aed0c4e0.png)

**Comment by michael-mezgo** [comment]
Generell würde ich mich sehr freuen, wenn es in Zukunft mehr Möglichkeiten für Ersatz-Checkins geben würde (auch mit benutzerdefinierten Haltestellen - anhand von Koordinaten) und eigenen Zeiten.

Ich kann mir schon vorstellen, dass dies aus programmiertechnischer Sicht und aus Sicht der Datenbank schwierig werden könnte. Dennoch würde ich mich über eine Implementierung freuen (besonders, da aus Österreich nicht alle Verkehre in der API der DB hinterlegt sind). 
Es gibt einige (Express-)Regionalbuslinien, welche mehrere hundert Kilometer lang sind - über diese würde ich mich in der Träwelling-Statistik freuen ☺️.

**Comment by marhei** [comment]
I think the upcoming feature of "Manuelle Fahrten" (#2034) is a better solution than this "Ersatzcheckin" workaround.
